### ISSUE TITLE
Timeline component BEM class consistency and accessibility robustness

### ISSUE THREAD

**Issue by jonhurrell** [dwp/design-system#469]
## Describe the issue
BEM modifier used incorrectly, should be a new element context set on list items e.g.: `dwp-timeline__events-item`.

### Accessibility
We need to look at the use of headings for the events item title, with potential hierarchy conflict of using similar titles with the same heading level e.g.: accepted, declined, accepted. 
There's not enough context to be able to navigate between the different heading values as suggested by the example. This could be resolved as guidance in the accessibility section to reference `Don't use headings if multiple heading values are the same` in which case I think we should bake more options into the component to improve it's robustness, e.g. offer both semantic heading options for use in cases where title values are distinct and unique or where that's not possible a version that improves the coupling of content to a pseudo heading (perhaps) to aid non sighted navigation.
